util/log/clog_test.go

```diff
@@ -355,41 +353,70 @@ func TestGetLogReader(t *testing.T) {
 	}
 	warnName := filepath.Base(warn.file.Name())
 
+	curDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	relPath, err := filepath.Rel(curDir, warn.file.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	otherFile, err := os.Create(filepath.Join(*logDir, "other.txt"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	otherFile.Close()
+
 	testCases := []struct {
-		filename string
-		allowAbs bool
-		expErr   bool
+		filename           string
+		expErrRestricted   string
+		expErrUnrestricted string
 	}{
 		// File is not specified (trying to open a directory instead).
-		{*logDir, false, true},
-		{*logDir, true, true},
+		{*logDir, "pathnames must be basenames", "not a regular file"},
 		// Absolute filename is specified.
-		{warn.file.Name(), false, true},
-		{warn.file.Name(), true, false},
-		// File not matching log RE.
-		{"cockroach.WARNING", false, true},
-		{"cockroach.WARNING", true, true},
-		{filepath.Join(*logDir, "cockroach.WARNING"), false, true},
-		{filepath.Join(*logDir, "cockroach.WARNING"), true, true},
-		// Relative filename is specified.
-		{warnName, true, false},
-		{warnName, false, false},
+		{warn.file.Name(), "pathnames must be basenames", ""},
+		// Symlink to a log file.
+		{filepath.Join(*logDir, "logtest.WARNING"), "pathnames must be basenames", ""},
+		// Symlink relative to logDir.
+		{"logtest.WARNING", "malformed log filename", ""},
+		// Non-log file.
+		{"other.txt", "malformed log filename", "malformed log filename"},
+		// Non-existent file matching RE.
+		{"cockroach.roach0.root.log.ERROR.2015-09-25T19_24_19Z.1", "no such file", "no such file"},
+		// Base filename is specified.
+		{warnName, "", ""},
+		// Relative path with directory components.
+		{relPath, "pathnames must be basenames", ""},
 	}
 
 	for i, test := range testCases {
-		reader, err := GetLogReader(test.filename, test.allowAbs)
-		if (err != nil) != test.expErr {
-			t.Errorf("%d: expected error %t; got %t: %s", i, test.expErr, err != nil, err)
-		}
-		if reader != nil {
-			reader.Close()
+		for restricted, expErr := range []string{test.expErrUnrestricted, test.expErrRestricted} {
+			reader, err := GetLogReader(test.filename, restricted == 1)
```

Comment:
perhaps `s/restricted == 1/expErr == test.expErrRestricted/` would be explicit enough here